File: решения/java/src/Lab14/task3185/Point.java
package Lab14.task3185;

/**
 * @author akryukov
 *         13.07.2017
 */
public class Point {
    public int x;
    public int y;
    public double distanceTo(Point p){
        int dx = p.x - x;
        int dy = p.y - y;
        return Math.sqrt(dx*dx + dy*dy);
    }
}

File: решения/cs/Classes/task3185/Point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Classes.task3185
{
    public class Point
    {
        public int x;
        public int y;

        public double DistanceTo(Point p)
        {
            if(p == null)
            {
                throw new ArgumentException("Конечная точка не инициализирована");
            }
            int dx = p.x - x;
            int dy = p.y - y;
            return Math.Sqrt(dx * dx + dy * dy);
        }
    }
}
